components/nocodb/nocodb.app.mjs

```diff
@@ -1,11 +1,146 @@
+import { Api } from "nocodb-sdk";
+
 export default {
   type: "app",
   app: "nocodb",
-  propDefinitions: {},
+  propDefinitions: {
+    projectId: {
+      type: "string",
+      label: "Project Id",
+      description: "The id of the project",
+      async options() {
+        const { list } = await this.listProjects();
+        return list.map((project) => ({
+          label: project.title,
+          value: project.id,
+        }));
+      },
```

Comment:
You can use the `page` or `prevContext` prop to handle pagination for `projectId` prop here.

Reply:
@michelle0927 I talked to nocodb support, they don't have pagination to this endpoint

Reply:
Aren't these the endpoints being used by the sdk?
List Projects - https://all-apis.nocodb.com/#tag/Project/operation/project-list
List Tables - https://all-apis.nocodb.com/#tag/DB-table/operation/db-table-list

Reply:
yes they are, but the pagination in 'list-projects' doesn't work, so I talked to support and they said this really doesn't have pagination